acstools/acszpt.py

```diff
@@ -268,116 +259,92 @@ def _check_date(self, fmt='%Y-%m-%d'):
 
         Returns
         -------
-        tab : `astropy.table.QTable` or `None`
-            If the request was successful, returns a table; otherwise, `None`.
-
+        table : `astropy.table.QTable` or `None`
+            If the request was successful, returns an astropy Qtable; otherwise, `None`.
         """
-        LOG.info('Checking inputs...')
-        valid_inputs = self._check_inputs()
-
-        if valid_inputs:
-            LOG.info(f'Submitting request to {self._url}')
-            self._submit_request()
-            if self._failed:
-                return
 
-            LOG.info('Parsing the response and formatting the results...')
-            self._parse_and_format()
-            for w in self._warnings:
-                LOG.warning(w)
+        # check user input date, detector, and filter
+        bool_inputs = self._check_inputs()
+
+        if bool_inputs:
+
+            # select between all filters or a single filter depending on if user has
+            # specified a filter, and then generate a request body to send the API
+            if self.filt is None:
+                # URL to invoke all_filter API on AWS APIGateway
+                invokeURL = "https://vtbopx9sf3.execute-api.us-east-1.amazonaws.com/main/all-filter-resource"  # noqa
+                # Generate a request body to send the API
+                body = '{"date": "%s", "detector": "%s"}' % (self.date, self.detector)
+            else:
+                # URL to invoke single_filter API on AWS APIGateway
+                invokeURL = "https://vtbopx9sf3.execute-api.us-east-1.amazonaws.com/main/single-filter-resource"  # noqa
+                # Generate a request body to send the API
+                body = '{"date": "%s", "detector": "%s", "filter": "%s"}' % (self.date, self.detector, self.filt)
+
+            # send request to APIGateway with try/except clauses
+            help_desk_string = ("\nIf this error persists, please contact the HST Help Desk at \n"
+                                "https://stsci.service-now.com/hst")
+
+            request_exception = True
+
+            try:
+                response = requests.post(invokeURL, data=body)
+                response.raise_for_status()
+
+            except requests.exceptions.HTTPError:
+                err_string = "HTTP Error to AWS API Gateway:" + help_desk_string
+                print(err_string)
+            except requests.exceptions.ConnectionError:
+                err_string = "Error Connecting to AWS API Gateway:" + help_desk_string
+                print(err_string)
+            except requests.exceptions.Timeout:
+                print("Timeout Error to AWS API Gateway:" + help_desk_string)
+            except requests.exceptions.RequestException:
+                print("Request Exception Error to AWS API Gateway:" + help_desk_string)
+            else: 
+                request_exception = False
+
+            if request_exception:
+                sys.exit()
```

Comment:
In a situation like this, you might want to consider writing your own exception class, for example:

```python
class ACSZeropointQueryError(Exception):
    pass
```

And then use that like this:

```suggestion
            except requests.exceptions.HTTPError as err:
                raise ACSZeropointQueryError(f"HTTP Error to AWS API Gateway:{help_desk_string}") from err
            except requests.exceptions.ConnectionError as err:
                raise ACSZeropointQueryError(f"Error Connecting to AWS API Gateway:{help_desk_string}") from err
            except requests.exceptions.Timeout as err:
                raise ACSZeropointQueryError(f"Timeout Error to AWS API Gateway:{help_desk_string}") from err
            except requests.exceptions.RequestException as err:
                raise ACSZeropointQueryError(f"Request Exception Error to AWS API Gateway:{help_desk_string}") from err
```

Reply:
this provides me with the same output/net effect as the screenshot above, I will implement it

Reply:
actually scratch the above comment, the traceback with this is much longer

Reply:
You could also do `from None` if you want.

Reply:
I think I'll make that swap, yes